src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,83 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { User } from '../../types';
 import { Loader } from '../Loader';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { getUser } from '../../api';
+import { clearTodo } from '../../features/currentTodo';
 
 export const TodoModal: React.FC = () => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [user, setUser] = useState<User | null>(null);
+
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const dispatch = useAppDispatch();
+
+  useEffect(() => {
+    setTimeout(() => {
+      const fetchUser = async () => {
+        if (currentTodo) {
+          try {
+            const fetchedUser = await getUser(currentTodo.userId);
+
+            setUser(fetchedUser);
+          } catch {
+            setUser(null);
+          } finally {
+            setIsLoading(false);
+          }
+        }
+      };
+
+      fetchUser();
+    }, 500);
+  }, [currentTodo]);
```

Comment:
The `useEffect` hook depends on `currentTodo`, which is correct. However, ensure that `setIsLoading(true)` is called at the beginning of the effect to reset the loading state when `currentTodo` changes. This will ensure the loader is displayed while fetching new user data.